hack/k8s-minikube.sh

```diff
@@ -99,16 +107,16 @@ Valid options:
       Enable logging of debug messages from this script.
 
 The command must be either:
-  up:        starts the minikube cluster
-  down:      stops the minikube cluster
-  status:    gets the status of the minikube cluster
-  delete:    completely removes the minikube cluster VM destroying all state
-  docker:    information on the minikube docker environment
-  dashboard: enables access to the Kubernetes GUI within minikube
-  ingress:   enables access to the Kubernetes ingress URL within minikube
-  istio:     installs Istio into the minikube cluster
-  bookinfo:  installs Istio's bookinfo demo (make sure Istio is installed first)
-  gwurl:     displays the Ingress Gateway URL
+  up:                starts the minikube cluster
+  down:              stops the minikube cluster
+  status:            gets the status of the minikube cluster
+  delete:            completely removes the minikube cluster VM destroying all state
+  docker:            information on the minikube docker environment
+  dashboard:         enables access to the Kubernetes GUI within minikube
+  ingress:           enables access to the Kubernetes ingress URL within minikube
+  istio:             installs Istio into the minikube cluster
+  bookinfo:          installs Istio's bookinfo demo (make sure Istio is installed first)
+  gwurl [port name]: displays the Ingress Gateway URL - if port name is given, the gateway port is also shown.
```

Comment:
Is 'port name' literal? Or how do users know what value to supply?

Reply:
it is not literal - its the actual port name specified in the istio-gateway config. For example, you can get the gateway port for "http2-prometheus" (that's the port name you can see here being defined in the helm values file: https://github.com/istio/istio/blob/ce90c45dfe8aaf46315aca1b31ddbc2d12a4d090/install/kubernetes/helm/subcharts/gateways/values.yaml#L53) - thus getting the IP:port needed to access Prometheus.

Reply:
Obviously, the names of these ports vary - depends on what the helm charts use, or what users add to the istio-gateway.

Reply:
Do we know what the 'default' is, so that we can suggest that name here? I guess we don't want to return arbitrary ingress-gw+ports by default , but Kiali - no?

Reply:
There is no "default" really - it depends what URL you want. For example, I do pass in "http2" to get the Bookinfo URL (because I know the bookinfo demo will be going over that port) - see https://github.com/kiali/kiali/blob/6e946ed231d31d9386f5cf70132b67a7b995852e/hack/k8s-minikube.sh#L178 
That way when you install bookinfo, the script will echo out the proper URL.

But, for example, the reason I did it this way is I need to know the URLs for **all** the different add-ons (not just kiali). There isn't a "default" there in that case so I needed a command to run that would get me these:

./k8s-minikube.sh gwurl http2-prometheus
./k8s-minikube.sh gwurl http2-grafana
./k8s-minikube.sh gwurl http2-tracing
./k8s-minikube.sh gwurl http-kiali

All of those return different ports - and they only work if the istio gateway has those port names defined.

We could create another command "gwurldefaults" to spit those four out?